Solutions/Corvus.Extensions.Newtonsoft.Json/Corvus/Extensions/Json/Internal/JsonNetPropertyBagFactory.cs

```diff
@@ -62,14 +62,7 @@ public IPropertyBag Create(JObject jObject)
             IEnumerable<KeyValuePair<string, object>>? propertiesToSetOrAdd,
             IEnumerable<string>? propertiesToRemove)
         {
-            if (!(input is IJsonNetPropertyBag propertyBag))
-            {
-                throw new ArgumentException(
-                    $"The input property bag must be a {nameof(JsonNetPropertyBag)}",
-                    nameof(input));
-            }
-
-            IReadOnlyDictionary<string, object> existingProperties = propertyBag.AsDictionary();
+            IReadOnlyDictionary<string, object> existingProperties = input.AsDictionary();
```

Comment:
Does this do the same thing? The old `AsDictionary` would preserve the `JObject`-ness. I'm now wondering if scenarios where this works:

```cs
bag.TryGet("k", out MyCustomType result);
```

(i.e., deserialize to `MyCustomType`) and where this would have worked before:

```cs
IPropertyBag brandNewBag = bagFactory.CreateModified(bad, toAdd, null);
brandNewBag.TryGet("k", out CustomType result);
```

I'm not sure what will happen in `CreateModified`. It's going to try to construct a new `JsonPropertyBag` with the ctor that takes an `IDictionary<string, object>`, and all the nested object type entries in that dictionary will be an `IPropertyBag`. Maybe that's OK because it will actually end up hitting the `PropertyBagConverter`, which will extract the `JObject` that's in that nested property bag, and perhaps it will all just work. But I think we definitely need specs that check that `CreateModified` works correctly for transferring, adding, and removing nested object type properties.

Reply:
>  we definitely need specs that check that CreateModified works correctly for transferring, adding, and removing nested object type properties

Assuming you mean "adding and removing properties that are objects", that's fine. I've added a spec for that.